confignode/src/main/java/org/apache/iotdb/confignode/manager/load/balancer/RouteBalancer.java

```diff
@@ -241,7 +256,81 @@ public void stopRouteBalancingService() {
   }
 
   private void balancingRegionLeader() {
-    // TODO: IOTDB-4768
+    balancingRegionLeader(TConsensusGroupType.SchemaRegion);
+    balancingRegionLeader(TConsensusGroupType.DataRegion);
+  }
+
+  private void balancingRegionLeader(TConsensusGroupType regionGroupType) {
+    // Collect latest data to generate leaderBalancer
+    MCFLeaderBalancer leaderBalancer =
+        new MCFLeaderBalancer(
+            getPartitionManager().getAllReplicaSetsMap(regionGroupType),
+            regionRouteMap.getRegionLeaderMap(),
+            getNodeManager()
+                .filterDataNodeThroughStatus(
+                    NodeStatus.Unknown, NodeStatus.ReadOnly, NodeStatus.Removing)
+                .stream()
+                .map(TDataNodeConfiguration::getLocation)
+                .map(TDataNodeLocation::getDataNodeId)
+                .collect(Collectors.toSet()));
+
+    // Calculate the optimal leader distribution
+    Map<TConsensusGroupId, Integer> leaderDistribution =
+        leaderBalancer.generateOptimalLeaderDistribution();
+
+    // Transfer leader to the optimal distribution
+    AtomicInteger requestId = new AtomicInteger(0);
+    AsyncClientHandler<TRegionLeaderChangeReq, TSStatus> clientHandler =
+        new AsyncClientHandler<>(DataNodeRequestType.CHANGE_REGION_LEADER);
+    leaderDistribution.forEach(
+        (regionGroupId, newLeaderId) -> {
+          if (newLeaderId != regionRouteMap.getLeader(regionGroupId)) {
+            String consensusProtocolClass;
+            switch (regionGroupId.getType()) {
+              case SchemaRegion:
+                consensusProtocolClass = SCHEMA_REGION_CONSENSUS_PROTOCOL_CLASS;
+                break;
+              case DataRegion:
+              default:
+                consensusProtocolClass = DATA_REGION_CONSENSUS_PROTOCOL_CLASS;
+                break;
+            }
+            LOGGER.info(
+                "[LeaderBalancer] Try to change the leader of Region: {} to DataNode: {} ",
+                regionGroupId,
+                newLeaderId);
+            changeRegionLeader(
```

Comment:
I have a question .   change region leader during removing datanode or adding datanode, Whether Scaling Is Affected ？

Reply:
No, the ConfigNode-leader just try to optimize the RegionGroups' leader distribution, and this process doesn't effect the validity of cluster extension.